src/lib.rs

```diff
@@ -3918,6 +4021,92 @@ mod tests {
         }
     }
 
+    mod dst_layout {
```

Comment:
IMO it'd be cleaner if you just put these in the parent module and called added a `test_dst_layout` prefix to their names. If we want to consider a broader overhaul of how these tests are structured, that's fine, but for now we should be consistent with the existing convention - IMO consistency is more important.